dnf/base.py

```diff
@@ -1631,7 +1634,8 @@ def distro_sync(self, pkg_spec=None):
         if pkg_spec is None:
             self._goal.distupgrade_all()
         else:
-            sltrs = dnf.subject.Subject(pkg_spec)._get_best_selectors(self.sack)
+            sltrs = dnf.subject.Subject(pkg_spec) \
+                ._get_best_selectors(self.sack, None, self.conf.obsoletes)
```

Comment:
Probably better would be: ._get_best_selectors(self.sack, obsoletes=self.conf.obsoletes)